cmd/ocitools/generate.go

```diff
@@ -463,3 +412,177 @@ func parseBindMount(s string) (string, string, string, error) {
 
 	return source, dest, options, nil
 }
+
+func addSeccomp(g *generate.Generator, context *cli.Context) error {
+
+	// Set the DefaultAction of seccomp
+	seccompDefault := context.String("seccomp-default")
+	err := g.SetDefaultSeccompAction(seccompDefault)
+	if err != nil {
+		return err
+	}
+
+	// Add the additional architectures permitted to be used for system calls
+	seccompArch := context.String("seccomp-arch")
+	architectureArgs := strings.Split(seccompArch, ",")
+	err = g.SetSeccompArchitectures(architectureArgs)
+	if err != nil {
+		return err
+	}
+
+	if context.IsSet("seccomp-kill") {
+		seccompKill := context.String("seccomp-kill")
+		killArgs := strings.Split(seccompKill, ",")
+		setSyscallArgsSlice := [][]string{}
+		for _, killArg := range killArgs {
+			comparisonArgs := strings.Split(killArg, ":")
+			if len(comparisonArgs) == 5 {
+				setSyscallArgs := []string{"kill",
+					comparisonArgs[0],
+					comparisonArgs[1],
+					comparisonArgs[2],
+					comparisonArgs[3],
+					comparisonArgs[4],
+				}
+				setSyscallArgsSlice = append(setSyscallArgsSlice, setSyscallArgs)
+			} else if len(comparisonArgs) == 1 {
+				setSyscallArgs := []string{"kill", comparisonArgs[0]}
+				setSyscallArgsSlice = append(setSyscallArgsSlice, setSyscallArgs)
+			} else {
+				return errors.New("Invalid syscall argument formatting")
+			}
+
+			for _, r := range setSyscallArgsSlice {
+				err := g.SetSyscallAction(r)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	if context.IsSet("seccomp-trace") {
+		seccompTrace := context.String("seccomp-trace")
```

Comment:
Looks like we have a repeated code pattern which could be abstracted into a helper function.